src/common/src/array/list_array.rs

```diff
@@ -161,6 +165,32 @@ impl ListArrayBuilder {
             self.value.append_datum(v);
         }
     }
+
+    /// Append a value to the current list array slot.
+    ///
+    /// You must call `finish_sub` later to finish the current list.
+    ///
+    /// ```
+    /// # use risingwave_common::array::ListArrayBuilder;
+    ///
+    /// let mut builder = ListArrayBuilder::with_type(10, DataType::List(Box::new(DataType::Int32)));
+    /// builder.append_sub(1);
+    /// builder.append_sub(2);
+    /// builder.append_sub(3);
+    /// builder.finish_sub(true);
+    /// let array = builder.finish();
+    /// assert_eq!(array.len(), 1);
+    /// ```
+    pub fn append_sub(&mut self, value: impl ToDatumRef) {
```

Comment:
Can we clean up these unnecessary functions?

Reply:
Sure